docs/extensibility-http.md

```diff
@@ -0,0 +1,525 @@
+    }
+}
+```
+
+Revise `./agent/src/main.rs`:
+
+```rust
+#[macro_use]
+extern crate failure;
+```
+
+Revise `./agent/Cargo.toml`:
+
+```TOML
+[dependencies]
+hyper-async = { version = "0.13.5", package = "hyper" }
+reqwest = "0.10.8"
+```
+
+## Akri Configuration CRD
+
+> **NOTE** Making this change means you must `helm install` a copy of this directory **not** Microsoft hosted
+
+Revise `./deployment/helm/crds/akri-configuration-crd.yaml`:
+
+```YAML
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: configurations.akri.sh
+spec:
+  group: akri.sh
+  versions:
+    - name: v0
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                protocol: # {{ProtocolHandler}}
+                  type: object
+                  properties:
+                    http: # {{HTTPDiscoveryHandler}}
+                      type: object
+                      properties:
+                        discoveryEndpoint:
+                          type: string
+...
+                  oneOf:
+                    - required: ["http"]
+```
+
+## Shared Configuration
+
+Revise `./shared/src/akri/configuration.rs`:
+
+```rust
+pub enum ProtocolHandler {
+    http(HTTPDiscoveryHandlerConfig),
+    ...
+}
+```
+
+And:
+
+```rust
+/// This defines the HTTP data stored in the Configuration
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct HTTPDiscoveryHandlerConfig {
+    pub discovery_endpoint: String,
+}
+```
+
+## Docker
+
+Revised `./dockerignore` to ensure `docker build ...` succeed:
+
+```console
+# Don't ignore these
+!samples/brokers/http
+!shared/
+```
+
+## Revise workspace Cargo file
+
+Revise `./Cargo.toml`:
+
+```TOML
+[workspace]
+members = [
+    ...
+    "samples/brokers/http",
+    ...
+]
+```
+
+> **NOTE ** This also ensures `http` inclusion using [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer)
+
+## Revise workspace Cross file
+
+Revise `./Cross.toml`:
+
+```TOML
+[target.x86_64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:x86_64-unknown-linux-gnu-0.1.16-0.0.6"
+
+[target.armv7-unknown-linux-gnueabihf]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:armv7-unknown-linux-gnueabihf-0.1.16-0.0.6"
+
+[target.aarch64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:aarch64-unknown-linux-gnu-0.1.16-0.0.6"
+```
+
+## Build Akri Agent|Controller
+
+```bash
+USER=[[GTHUB-USER]]
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make rust-crossbuild
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-agent
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-controller
+```
+
+## Build|Push Broker (HTTP)
+
+```bash
+HOST="ghcr.io"
+USER=[[GITHUB-USER]]
+REPO="akri" # Or your preferred GHCR repo prefix
```

Comment:
pretty sure i asked you to change this to `REPO="akri"`, but i think i was wrong.

maybe this should be `BROKER="http-broker"` and then BROKER should be used below in `docker build` command rather than REPO or REPO-broker?

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/60620b61185473dab04cd2fb77a3344905a2825b